metadata/src/main/java/org/apache/kafka/queue/EventQueue.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.queue;
+
+import org.slf4j.Logger;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+
+public interface EventQueue extends AutoCloseable {
+    interface Event {
+        void run() throws Exception;
+        default void handleException(Throwable e) {}
+    }
```

Comment:
> Let's document these two methods

Sure.  I will add some JavaDoc.

> Any thoughts on why this design instead of not allowing run to throw a checked exception? For example:

Checked exceptions are sort of a failed idea.  This would just force you to wrap all the checked exceptions that the event could throw, and unwrap them in handleException.  That would not help the clarity of the controller code.

Also, implementing Runnable isn't really useful here because Java 8 lambdas can be used on any single-function interface, not just Runnable.